server/repository/src/test_db/mod.rs

```diff
@@ -0,0 +1,109 @@
+#[cfg(not(feature = "postgres"))]
+pub(crate) mod sqlite;
+#[cfg(not(feature = "postgres"))]
+pub use self::sqlite::*;
+
+#[cfg(feature = "postgres")]
+mod postgres;
+#[cfg(feature = "postgres")]
+pub use self::postgres::*;
+
+use crate::{
+    database_settings::DatabaseSettings,
+    migrations::Version,
+    mock::{
+        insert_all_mock_data, insert_extra_mock_data, MockData, MockDataCollection, MockDataInserts,
+    },
+    StorageConnection, StorageConnectionManager,
+};
+
+/// Generic setup method to help setup test enviroment
+/// - sets up database (create one and initialises schema), drops existing database
+/// - creates connectuion
+/// - inserts mock data
+pub async fn setup_all(
+    db_name: &str,
+    inserts: MockDataInserts,
+) -> (
+    MockDataCollection,
+    StorageConnection,
+    StorageConnectionManager,
+    DatabaseSettings,
+) {
+    let result = setup_test(SetupOption {
+        db_name,
+        inserts,
+        ..Default::default()
+    })
+    .await;
+    (
+        result.core_data,
+        result.connection,
+        result.connection_manager,
+        result.db_settings,
+    )
+}
+
+pub async fn setup_all_with_data(
+    db_name: &str,
+    inserts: MockDataInserts,
+    extra_mock_data: MockData,
+) -> (
+    MockDataCollection,
+    StorageConnection,
+    StorageConnectionManager,
+    DatabaseSettings,
+) {
+    let result = setup_test(SetupOption {
+        db_name,
+        inserts,
+        extra_mock_data,
+        ..Default::default()
+    })
+    .await;
+    (
+        result.core_data,
+        result.connection,
+        result.connection_manager,
+        result.db_settings,
+    )
+}
+
+#[derive(Default)]
+pub struct SetupOption<'a> {
+    pub db_name: &'a str,
+    pub version: Option<Version>,
+    pub inserts: MockDataInserts,
+    pub extra_mock_data: MockData,
+}
+
+pub struct SetupResult {
+    pub core_data: MockDataCollection,
+    pub connection: StorageConnection,
+    pub connection_manager: StorageConnectionManager,
+    pub db_settings: DatabaseSettings,
+}
+
+// Object/Struct input/output allow adding new setup parameters without mass
+// refactor
+pub async fn setup_test<'a>(
+    SetupOption {
```

Comment:
Returning struct vs tuple is much easier to extend (if the consumer is destructuring), don't have to update consumer code.